src/modules/ekf2/EKF/mag_heading_control.cpp

```diff
@@ -68,18 +68,34 @@ void Ekf::controlMagHeadingFusion(const magSample &mag_sample, const bool common
 	if (_control_status.flags.yaw_align) {
 		// mag heading
 		aid_src.innovation = wrap_pi(getEulerYaw(_R_to_earth) - aid_src.observation);
+		_mag_heading_innov_lpf.update(aid_src.innovation);
+
+		if (!_control_status.flags.gps) {
+			_control_status.flags.mag_consistent = true;
```

Comment:
I know it makes the logic easier below, but it still seems like it would make more sense for mag_consistent to only become true if and when we're able to verify it.

Reply:
Ok, I think it makes sense. However I also decided to set it to consistent when there is a reset to mag. It would be strange to reset to mag and not declare it consistent, right? And if the yaw isn't initialized, we need to use the mag even if we don't know if it will be consistent with IMU/GNSS compassing.

Reply:
Got it, so the flag is simply stating that the mag heading is consistent with the filter yaw, not necessarily that it's been independently verified.